package/iframe/main.js

```diff
@@ -54,56 +54,38 @@
+            // value.
             true, // 'getMobileIdAssertion' in window.navigator || 'api.window.Navigator.getMobileIdAssertion',
             true, // 'manifest.precompile',
             ['hardware.memory', 512],
             ['hardware.memory', 1024],
             true, // NFC
-            'acl.version',
+            'acl.version', // OpenMobile ACL
             // Don't add any more as long as bug 1172487 is not fixed, it won't
             // work correctly.
         ];
 
         features.forEach(function(key) {
             if (typeof key === 'boolean') {
-                // Hardcoded boolean value, just pass it to promises directly.
+                // Hardcoded boolean value, just pass it directly.
                 promises.push(key);
-            }
-            else if (typeof key === 'string') {
-                if (key.substr(0, 9) === 'manifest.' ||
-                    key === 'acl.version') {
-                    // Handle cases where we need to call getFeature() instead
-                    // of just hasFeature().
-                    // It's necessary in 2 cases:
-                    // - The `manifest.*` properties ; Because of bug 1098470,
-                    //   hasFeature(manifest.*) can fail, while
-                    //   getFeature(manifest.*) will work. On later builds,
-                    //   where this bug was fixed, the behaviour is reversed
-                    //   so we call hasFeature() first and *then* getFeature()
-                    //   if it returned a falsy value.
-                    // - The `acl.version` property, which is only available
-                    //   through getFeature().
-                    promises.push(new Promise(function(resolve, reject) {
-                        navigator.hasFeature(key).then(function(supported) {
-                            if (!supported) {
-                                navigator.getFeature(key).then(function(feat) {
-                                    resolve(feat);
-                                });
-                            } else {
-                                resolve(supported);
-                            }
-                        });
-                    }));
-                } else {
-                    // Regular case: just call hasFeature().
+            } else if (typeof key === 'string') {
+                if (key === 'acl.version') {
+                    // This feature has a value, so we need to call
+                    // getFeature(), but we actually only care about the fact
+                    // that it's not empty, so we can just use the getFeature()
+                    // promise directly.
+                    promises.push(navigator.getFeature(key));
+                } else if (typeof window.hasFeature !== 'undefined') {
+                    // Regular boolean feature: just call hasFeature().
                     promises.push(navigator.hasFeature(key));
```

Comment:
This seems to no longer handle the case for bug 1098470 (check `getFeature` after `hasFeature`). Is that intentional?